main/src/cgeo/geocaching/gcvote/GCVote.java

```diff
@@ -42,6 +43,64 @@ private GCVote() {
         // utility class
     }
 
+    private static class SingletonHolder {
+        private final static GCVote INSTANCE = new GCVote();
+    }
+
+    public static GCVote getInstance() {
+        return SingletonHolder.INSTANCE;
+    }
+
+    public static StatusCode login() {
+        final ImmutablePair<String, String> login = Settings.getGCVoteLogin();
+
+        if (login == null) {
+            Log.e("Credentials can't be retreived");
+            return StatusCode.NO_LOGIN_INFO_STORED;
+        }
+
+        if (StringUtils.isEmpty(login.left) || StringUtils.isEmpty(login.right)) {
```

Comment:
If it's empty, it's blank, and that has been checked already in `Settings.getGCVoteLogin()`. Either it returns `null` or it returns potentially valid credentials.